src/main/java/fi/minedu/oiva/backend/service/LupaService.java

```diff
@@ -81,6 +82,8 @@
     public Collection<Lupa> getAll(final String... withOptions) { // TODO: Implement filter
         final SelectJoinStep<Record> query = baseLupaSelect();
         baseLupaFilter().ifPresent(query::where);
+        // filteröidään tulevat luvat (ja varmaan kohta vanhatkin)
+        query.where(LUPA.ALKUPVM.le(DSL.currentDate()));
```

Comment:
eikö tässä voi tulla virhe, koska edellinen rivi on voinut muodostaa where -lauseen jo, jolloin tässä pitäisi käyttää AND operaatiota.

Reply:
jos tuo where LUPA.ALKUPVM suodatus tulee päteä kaikkiin lupiin niin sen voisi sisällyttää baseLupaFilteriin